### ISSUE TITLE
Time out.

### ISSUE THREAD

**Issue by zsolt247** [cfjedimaster/brackets-w3cvalidation#32]
Hi, the W3C Validation stopped working, I get the following message:

"W3CValidation has timed out after waiting for 10000 ms"

Using the latest Brackets 1.3, everything is updated.

Your help would be appreciated. Thanks.

**Comment by cfjedimaster** [comment]
Wow, it looks like it stopped working. It was working fine earlier.

**Comment by cfjedimaster** [comment]
Fixed - will be on the registry in a bit.

**Comment by Olivaresp** [comment]
I have the same. 
And the w3c validation service blocks my ip due to abuse of the requests

**Comment by cfjedimaster** [comment]
I saw this happen Monday. I don't know how as I had only tested a few times. Will look again later this week.

**Comment by zsolt247** [comment]
I wouldn't mind if the validation was on demand only instead of automatically re-validating with every code change... Just a thought...

**Comment by cfjedimaster** [comment]
Unfortunately that's not how the Brackets Linting API works. I'd have to
change the extension to be more manual.

On Wed, Jun 24, 2015 at 8:42 AM, zsolt247 <notifications@github.com> wrote:

> I wouldn't mind if the validation was on demand only instead of
> automatically re-validating with every code change... Just a thought...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfjedimaster/brackets-w3cvalidation/issues/32#issuecomment-114871785>
> .
>



-- 
===========================================================================
Raymond Camden, Developer Advocate for MobileFirst at IBM

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by LukeMcGurdy** [comment]
Hi Raymond,

I'm seeing the same error message today.

**Comment by cfjedimaster** [comment]
If you can, open up the dev tools and see if you are being blocked for
hitting the API too much.

On Tue, Jul 14, 2015 at 9:07 AM, LukeMcGurdy <notifications@github.com>
wrote:

> Hi Raymond,
>
> I'm seeing the same error message today.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfjedimaster/brackets-w3cvalidation/issues/32#issuecomment-121248722>
> .
>



-- 
===========================================================================
Raymond Camden, Developer Advocate for MobileFirst at IBM

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by marcusyoung** [comment]
I am having the same problem today also. Only installed the extension yesterday, when it worked fine.

 Failed to load resource: the server responded with a status of 403 (Forbidden)

**Comment by cfjedimaster** [comment]
Yep - nothing I can do about it for now - their API is blocking the
extension for misuse even though it shouldn't be hitting it that much. I've
seen an example of using the Java library from Node, but that would require
a big rewrite of the extension. For now, I'm open to someone giving me a
hand here because I'm out of ideas.

On Thu, Jul 16, 2015 at 8:17 AM, Marcus Young <notifications@github.com>
wrote:

> I am having the same problem today also. Only installed the extension
> yesterday, when it worked fine.
>
> Failed to load resource: the server responded with a status of 403
> (Forbidden)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfjedimaster/brackets-w3cvalidation/issues/32#issuecomment-121953069>
> .
>



-- 
===========================================================================
Raymond Camden, Developer Advocate for MobileFirst at IBM

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by LukeMcGurdy** [comment]
Hi Raymond,

As of this moment it looks fine so there is nothing to check. I will add that a couple of days back my IP was blocked from the W3C validation site for 24 hours because I had apparently exceeded the max number of requests. I had be saving a lot that day so if I understand your response about the nature of the Brackets Linting API this might explain why my IP was blacklisted.

I appreciate the extension all the same.

**Comment by jbwCode** [comment]
Help! I have been using this extension fine for a couple of days with no problems, but suddenly I received a timeout "IP blocked due to re-requesting files too often"  The specific type of abuse we observed is: re-requesting the same resource too frequently. Specifically, we received at least 500 requests for the same resource (URI) from your IP address within a ten-minute time interval. http://www.w3.org/Help/abuse-info/re-reqs.html

**Comment by cfjedimaster** [comment]
Yep, known issue. Nothing I can really do about it for now. I've looked
into converting the extension to run a Jar locally, but it is a lot of work
I'm not sure I want to actually do at this moment.

On Sat, Aug 29, 2015 at 6:04 PM, jbwCode <notifications@github.com> wrote:

> Help! I have been using this extension fine for a couple of days with no
> problems, but suddenly I received a timeout "IP blocked due to
> re-requesting files too often" The specific type of abuse we observed is:
> re-requesting the same resource too frequently. Specifically, we received
> at least 500 requests for the same resource (URI) from your IP address
> within a ten-minute time interval.
> http://www.w3.org/Help/abuse-info/re-reqs.html
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfjedimaster/brackets-w3cvalidation/issues/32#issuecomment-136059685>
> .
>



-- 
===========================================================================
Raymond Camden, Developer Advocate for MobileFirst at IBM

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by jbwCode** [comment]
Hmmm... sad to hear, it's an awesome extension!  Maybe something in the html file is triggering the bug? I was using it fine with files in one folder for two days. When I opened up a folder with tutorials from a couple of months ago, it triggered the error...

**Comment by cfjedimaster** [comment]
Not sure it is a bug per se - it uses the built in linting API for
Brackets. Maybe it is triggering on a bunch of files at once? So you
literally got this just opening a folder?


On Sat, Aug 29, 2015 at 6:11 PM, jbwCode <notifications@github.com> wrote:

> Hmmm... sad to hear, it's an awesome extension! Maybe something in the
> html file is triggering the bug? I was using it fine with files in one
> folder for two days. When I opened up a folder with tutorials from a couple
> of months ago, it triggered the error...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfjedimaster/brackets-w3cvalidation/issues/32#issuecomment-136061815>
> .
>



-- 
===========================================================================
Raymond Camden, Developer Advocate for MobileFirst at IBM

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by jbwCode** [comment]
One thing I can tell you, is that the files in the folder I just opened were missing "<!doctype html>"

**Comment by cfjedimaster** [comment]
No promises, but this week Ill try to see if I can see Brackets doing
something weird network wise. 500 seems like a lot in ten minutes.

On Sat, Aug 29, 2015 at 6:15 PM, jbwCode <notifications@github.com> wrote:

> One thing I can tell you, is that the files in the folder I just opened
> were missing "<!doctype html>"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfjedimaster/brackets-w3cvalidation/issues/32#issuecomment-136061934>
> .
>



-- 
===========================================================================
Raymond Camden, Developer Advocate for MobileFirst at IBM

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by jbwCode** [comment]
Thanks, I was thinking of blocking Brackets on my Windows 7 firewall to prevent spurious packets from getting out +1:

**Comment by cfjedimaster** [comment]
Just easier to disable my extension. :)

On Sat, Aug 29, 2015 at 6:20 PM, jbwCode <notifications@github.com> wrote:

> Thanks, I was thinking of blocking Brackets on my Windows 7 firewall to
> prevent spurious packets from getting out +1:
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfjedimaster/brackets-w3cvalidation/issues/32#issuecomment-136062083>
> .
>



-- 
===========================================================================
Raymond Camden, Developer Advocate for MobileFirst at IBM

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by jbwCode** [comment]
Looks like a good opportunity to request a feature from the Brackets team, to allow a manual Brackets Linting API ... a feature like this this would be best served on demand like "remove white space on save".

**Comment by jbwCode** [comment]
Noticed that another difference in the troublemaker files is that the html had inline style tags, unlike the earlier html files that gave no trouble. The good ones used the link tag to a stylesheet css file ...

**Comment by cfjedimaster** [comment]
So, I did a bit of checking, and as far as I can see, Brackets isn't
running my validation script overly much. I opened a folder with a lot of
files, and it didn't scan everything- just the HTML files I opened.

On Sat, Aug 29, 2015 at 6:42 PM, jbwCode <notifications@github.com> wrote:

> Noticed that another difference in the troublemaker files is that the html
> had inline style tags, unlike the earlier html files that gave no trouble.
> The good ones used the link tag to a stylesheet css file ...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfjedimaster/brackets-w3cvalidation/issues/32#issuecomment-136062922>
> .
>



-- 
===========================================================================
Raymond Camden, Developer Advocate for MobileFirst at IBM

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by jbwCode** [comment]
Thanks for checking! IMHO, to receive over 500 hits in a few minutes would imply a loop somewhere. I just reproduced the error (my IP now blocked for another 24 hours) with this html (I deliberately left out <!doctype html>, <title>, and <meta>, in order to generate some normal validation problems...):

<html>

<head>


  <style type="text/css">
    body {
      background-color: #d2b48c;
      margin-left: 20%;
      margin-right: 20%;
      border: 2px dotted white;
      padding: 10px 10px 10px 10px;
      font-family: sans-serif;
    }

  </style>
</head>

<body>
  <h1>Starbuzz Coffee Beverages</h1>

  <h2>House Blend, $1.49</h2>
  <p id="houseblend">A Smooth, mild blend of coffees from Mexico, Bolivia, and Guatemala.</p>

  <h2>Mocha Cafe Latte, $2.35</h2>
  <p>Espresso, Steamed milk and chocolate syrup.</p>

  <h2>Cappuccino, $1.89</h2>
  <p>A mixture of espresso, steamed milk and foam.</p>

  <h2 id="chai">Chai Tea, $1.85</h2>
  <p>A spicy tea drink made with black tea, spices, milk and honey.
  </p>
  <p>
    <a href="mission.html" title="Read more about Starbuzz Coffee's important mission">our Mission</a>
    <br> Read the <a target="_blank" href="http://wickedlysmart.com/buzz/#Coffee" title="Read all about caffeine on the Buzz">Caffeine Buzz</a>
  </p>
</body>

</html>

**Comment by cfjedimaster** [comment]
Trying the same file, I don't see anything weird. It returns problems from
the API of course, but it isn't "hitting it" lots of times.

On Mon, Aug 31, 2015 at 9:33 AM, jbwCode <notifications@github.com> wrote:

> Thanks for checking! IMHO, to receive over 500 hits in a few minutes would
> imply a loop somewhere. I just reproduced the error (my IP now blocked for
> another 24 hours) with this html (I deliberately left out <!doctype html>,
> , and <meta>, in order to generate some normal validation problems...):</p>
> <p><html></p> <p><head></p> <p><style type="text/css"><br> body {<br>
> background-color: #d2b48c;<br> margin-left: 20%;<br> margin-right: 20%;<br>
> border: 2px dotted white;<br> padding: 10px 10px 10px 10px;<br>
> font-family: sans-serif;<br> }</p> <p></style><br> </head></p>
> <p><body><br> <h1>Starbuzz Coffee Beverages</h1></p> <p><h2>House Blend,
> $1.49</h2><br> <p id="houseblend">A Smooth, mild blend of coffees from
> Mexico, Bolivia, and Guatemala.</p></p> <p><h2>Mocha Cafe Latte,
> $2.35</h2><br> <p>Espresso, Steamed milk and chocolate syrup.</p></p>
> <p><h2>Cappuccino, $1.89</h2><br> <p>A mixture of espresso, steamed milk
> and foam.</p></p> <p><h2 id="chai">Chai Tea, $1.85</h2><br> <p>A spicy tea
> drink made with black tea, spices, milk and honey.<br> </p><br> <p><br> <a
> href="mission.html" title="Read more about Starbuzz Coffee's important
> mission">our Mission</a><br> <br> Read the <a target="_blank" href="
> http://wickedlysmart.com/buzz/#Coffee" title="Read all about caffeine on
> the Buzz">Caffeine Buzz</a><br> </p><br> </body></p> <p></html></p>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfjedimaster/brackets-w3cvalidation/issues/32#issuecomment-136389774>
> .
>



-- 
===========================================================================
Raymond Camden, Developer Advocate for MobileFirst at IBM

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by bamboorocket** [comment]
I just downloaded today and am getting same error:

W3CValidation has timed out after waiting for 10000 ms	

It worked a couple of times, but now keep getting this error.

I am on Mac OSX 10.11.1
Running latest Brackets - Release 1.5 build 1.5.0-16538 (release cf9cf4698)

**Comment by Snukii** [comment]
Getting the too many request problem too.

**Comment by cfjedimaster** [comment]
As I said, i was never able to reproduce the "hit the API a lot for one
file". If you can, let me know. Otherwise, there isn't much I can do about
it. I don't use Brackets much anymore unfortunately.

On Tue, Dec 15, 2015 at 5:47 AM, Snukii <notifications@github.com> wrote:

> Getting the too many request problem too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfjedimaster/brackets-w3cvalidation/issues/32#issuecomment-164740237>
> .
>



-- 
===========================================================================
Raymond Camden, Developer Advocate for MobileFirst at IBM

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by jbwCode** [comment]
This extension sounded like a very convenient time saver tool, and I thank the author for trying. But I have removed it, since I could not use it without getting my IP blocked. I have gone back to pasting my code into the W3C validation web pages. According to the author, the extension relies on a Brackets API. So something is not playing well together. Maybe the Brackets team can figure it out...

**Comment by cfjedimaster** [comment]
So I added a quick console msg to let me know every time it hits the API.
Again, I'm not seeing anything weird. I open an HTML file - it runs. I edit
- but don't save - and nothing happens. I save - it runs as expected.

If one of yall who hit the limit would like to test, I can share this mod
with you. Even if you are currently blocked, you can still see if the
logging is showing a lot of calls.

On Tue, Dec 15, 2015 at 9:45 AM, jbwCode <notifications@github.com> wrote:

> This extension sounded like a very convenient time saver tool, and I thank
> the author for trying. But I have removed it, since I could not use it
> without getting my IP blocked. I have gone back to pasting my code into the
> W3C validation web pages. According to the author, the extension relies on
> a Brackets API. So something is not playing well together. Maybe the
> Brackets team can figure it out...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfjedimaster/brackets-w3cvalidation/issues/32#issuecomment-164803437>
> .
>



-- 
===========================================================================
Raymond Camden, Developer Advocate for MobileFirst at IBM

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by jbwCode** [comment]
Just a thought here,  but what if the extension only hits the API once, but there is a condition where Brackets goes into a loop that generates multiple HTTP requests?

**Comment by cfjedimaster** [comment]
Sure - but how? Right now it uses the built in linter api which I can't
control, but as far as I can see, it is working right. On edit, do nothing
- but on save, run the lint.

If we can just find a way to recreate the "run a lot!" we would be golden.
;)

It is also possible the API is not telling the truth.


On Tue, Dec 15, 2015 at 9:58 AM, jbwCode <notifications@github.com> wrote:

> Just a thought here, but what if the extension only hits the API once, but
> there is a condition where Brackets goes into a loop that generates
> multiple HTTP requests?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfjedimaster/brackets-w3cvalidation/issues/32#issuecomment-164807286>
> .
>



-- 
===========================================================================
Raymond Camden, Developer Advocate for MobileFirst at IBM

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by JulienJo** [comment]
Hi! Getting the too many requests problem too. Did you updated the plugin with your mod?

**Comment by JulienJo** [comment]
Hum... It works again :-S

**Comment by cfjedimaster** [comment]
The block by w3c is temporary so it makes sense it would return. And no - I
have not done anything yet on the conversion required to run the validation
locally. As I think I said - it is possible if we call the Java code via
Node (I've seen an example of an extension doing that), but I've not had
the time to make that change.

On Wed, Jan 20, 2016 at 3:57 AM, Julien Jolly <notifications@github.com>
wrote:

> Hum... It works again :-S
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfjedimaster/brackets-w3cvalidation/issues/32#issuecomment-173150794>
> .
>



-- 
===========================================================================
Raymond Camden, Developer Advocate for MobileFirst at IBM

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by JulienJo** [comment]
Nevermind, it seems to work great again

**Comment by cfjedimaster** [comment]
To be clear, this is an issue and it *will* hit you again. It's just that
the block is temporary.

On Wed, Jan 20, 2016 at 9:07 AM, Julien Jolly <notifications@github.com>
wrote:

> Nevermind, it seems to work great again
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfjedimaster/brackets-w3cvalidation/issues/32#issuecomment-173230975>
> .
>



-- 
===========================================================================
Raymond Camden, Developer Advocate for MobileFirst at IBM

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by Bray90820** [comment]
I just installed the extension today and seem to also be getting the timeout error it worked for about 5 minutes

**Comment by cfjedimaster** [comment]
Yep, to be clear, I don't think anyone needs to confirm this anymore
(grin), it is definitely a bug. :)

On Mon, Jan 25, 2016 at 3:00 PM, Bray90820 <notifications@github.com> wrote:

> I just installed the extension today and seem to also be getting the
> timeout error it worked for about 5 minutes
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfjedimaster/brackets-w3cvalidation/issues/32#issuecomment-174661383>
> .
>



-- 
===========================================================================
Raymond Camden, Developer Advocate for MobileFirst at IBM

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by Bray90820** [comment]
Is a fix coming?

**Comment by cfjedimaster** [comment]
As it is a big change, no, not any time soon. I want to, but honestly, as I
don't use Brackets day to day anymore, it isn't high on my priority list.
Sorry to say that and just being honest.

On Mon, Jan 25, 2016 at 3:09 PM, Bray90820 <notifications@github.com> wrote:

> Is a fix coming?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfjedimaster/brackets-w3cvalidation/issues/32#issuecomment-174663608>
> .
>



-- 
===========================================================================
Raymond Camden, Developer Advocate for MobileFirst at IBM

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by Bray90820** [comment]
Alright thanks for the info

**Comment by jbwCode** [comment]
Did you find a better alternative to Brackets, or are you working with something other than HTML/CSS/Javascript?

**Comment by cfjedimaster** [comment]
I'm using Visual Studio Code now. I won't say it is better, but I will say
it is better for me.

On Mon, Jan 25, 2016 at 3:20 PM, jbwCode <notifications@github.com> wrote:

> Did you find a better alternative to Brackets, or are you working with
> something other than HTML/CSS/Javascript?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cfjedimaster/brackets-w3cvalidation/issues/32#issuecomment-174667196>
> .
>



-- 
===========================================================================
Raymond Camden, Developer Advocate for MobileFirst at IBM

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by jbwCode** [comment]
Hi Raymond, thanks for your input! I did not mean to open a debate on the virtues of Brackets :-)  There are plenty of other discussions on the web...

I will take a look at VSCode.  After looking at your website, reading some of your posts, and seeing your involvement in various advanced projects, it becomes apparent that your requirements go way beyond a client-side web editor. That is why I inquired as to what you are using now.  Very cool! 

I have found Brackets to be the ultimate code editor for Front End Development (and appears to be limited to that scope). which is my focus for now. But I do wish to eventually move on to full stack, back end, as well as mobile development. I have tried Notepad++, Sublime, Atom, etc...

I presume that it is the holy grail to find one code editor that can do it all... but an open source, cross platform, extensible, code editor that you can customize for all of your needs is going to be the best option.  I look to the future to decide what can best fill that role :-)

**Comment by nervousjuice** [comment]
Isnt Brackets getting incorporated into Dreamweaver?

**Comment by cfjedimaster** [comment]
It is being incorporated into DW, but not moving away from working by itself as well. More info here: http://blog.brackets.io/2015/10/05/update-from-the-adobe-brackets-team/
